Dockerfile.dapper

```diff
@@ -2,7 +2,7 @@ ARG RANCHER_VERSION=v2.8-head
 
 FROM rancher/rancher:$RANCHER_VERSION as rancher
 
-FROM registry.suse.com/bci/golang:1.20
+FROM registry.suse.com/bci/golang:1.21-2.2.25
```

Comment:
The version in rancher is set to [1.21](https://github.com/rancher/rancher/blob/release/v2.8/Dockerfile.dapper#L1), any reason why these would be different? I doubt it would affect the tests but just curious.

Reply:
The tag 1.21 is dynamic and always points to the latest tag in the 1.21 release line, which can potentially cause disparities between builds. But as you pointed out, it has been used in rancher/rancher which indicates it is acceptable. I think both tags should work, so I will keep it for now to avoid another round of reviews.